stdlib/Option.dora

```diff
@@ -0,0 +1,30 @@
+// FIXME: Identity/Equals/Stringable constraints should all end up on the individual methods that need them, not on the class
+// After nil is gone, this class needs some more intrinsics, so that it can treat <nullptr> as an empty Option (and get rid
+// of the isPresent field). Usual caveats apply: this only works for the first "layer", not for Options inside Options, etc.
+class Option[T : Identity + Equals + Stringable](let value: T, let isPresent: Bool) {
```

Comment:
Do you think we should implement this using subclassing? This should save some memory.

Reply:
Considering that I want to get rid of `nil`, I think I wouldn't bother with it. :-)